controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -95,6 +114,124 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+
+		// Create the StorageClass (StorageClass.Name == StorageClassClaim.Name) with StorageClassClaim as its owner.
+		_, err = controllerutil.CreateOrUpdate(ctx, r.Client, storageClass, func() error {
+
+			// Update the StorageClassClaim status.
+			r.updateStatus(ctx, r.storageClassClaim, ocsv1alpha1.StorageClassClaimPending)
+
+			// Check if finalizers are present, if not, add them.
+			if !contains(r.storageClassClaim.GetFinalizers(), StorageClassClaimFinalizer) {
+				storageClassClaimRef := klog.KRef(r.storageClassClaim.Name, r.storageClassClaim.Namespace)
+				log.Info("Finalizer not found for StorageClassClaim. Adding finalizer.", "StorageClassClaim", storageClassClaimRef)
+				r.storageClassClaim.ObjectMeta.Finalizers = append(r.storageClassClaim.ObjectMeta.Finalizers, StorageClassClaimFinalizer)
+				if err := r.Client.Update(ctx, r.storageClassClaim); err != nil {
+					log.Error(err, "Failed to update StorageClassClaim with finalizer.", "StorageClassClaim", storageClassClaimRef)
+					return err
+				}
+			}
+
+			// storageClassClaimStorageType is the storage type of the StorageClassClaim
+			var storageClassClaimStorageType providerpb.FulfillStorageClassClaimRequest_StorageType
+			switch r.storageClassClaim.Spec.Type {
+			case "blockpool":
+				storageClassClaimStorageType = providerpb.FulfillStorageClassClaimRequest_BLOCKPOOL + 0
+			case "sharedfilesystem":
+				storageClassClaimStorageType = providerpb.FulfillStorageClassClaimRequest_SHAREDFILESYSTEM + 2
+			}
+
+			// Update the StorageClassClaim status.
+			r.updateStatus(ctx, r.storageClassClaim, ocsv1alpha1.StorageClassClaimCreating)
+
+			// Call the `FulfillStorageClass` service on the provider server with StorageClassClaim as a request message.
+			// This call does nothing if the StorageClassClaim is already fulfilled.
+			// Wait (for `timeout` duration) to receive an acknowledgement from the provider server.
+			_, err = providerClient.FulfillStorageClassClaim(ctx, storagecluster.Status.ExternalStorage.ConsumerID, r.storageClassClaim.Name, r.storageClassClaim.Spec.EncryptionMethod, storageClassClaimStorageType)
+			if err != nil {
+				return err
+			}
+
+			/*	TODO
+				Periodically poll the provider server for resources provisioned for that claim.
+					Synchronous RPC since we need to wait for ACK to continue further
+					rpc PollProviderServer (PollProviderRequest) returns (PollProviderServerResponse) {}
+				Create (or do nothing if already created) the resources received from the provider server on the consumer cluster.
+			*/
+
+			// storageClass with the same name as storageClassClaim is controlled by the storageClassClaim.
+			// storageClass must be garbage collected when the respective storageClassClaim is deleted.
+			return controllerutil.SetControllerReference(r.storageClassClaim, storageClass, r.Scheme)
+		})
+
+		// Update the StorageClassClaim status.
+		r.updateStatus(ctx, r.storageClassClaim, ocsv1alpha1.StorageClassClaimReady)
```

Comment:
See the last comment about status update inside reconcile

```suggestion
		r.storageClassClaim =  ocsv1alpha1.StorageClassClaimReady
```